### DISCUSSION TITLE
Semantic Commits Type fails to `chore` when a dependency is sandwitched between devdependencies using group:allNonMajor

### DISCUSSION THREAD

**Discussion by carboneater** [renovatebot/renovate#17159] [Request Help]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, tell us what version of Renovate you run.

(Also reproduced using latest npm version of renovate)

### Please select which platform you are using if self-hosting.

Bitbucket Server

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

So I created this small reproduction repository: https://github.com/carboneater/renovate-semantic-reproduction

If you have some dependencies updated, then devDependencies, the semantic type results to `fix` which is as expected.
If there is a devDependency which gets sorted alphabetically before any dependency, the semantic type will be `chore`. (Failing to release a version and yada yada...)

You can have a look at https://github.com/carboneater/renovate-semantic-reproduction/pull/6
in `c434637`, `@alloy/is` wasn't in the dependency list.
in `e81dabb`, `@alloy/is` was added, and the commit type devolved from `fix` to `chore`

### Relevant debug logs

Sorry, I couldn't track the part where the commit message is built.

We do see the PR name changing though

https://gist.github.com/carboneater/1a5db51c94dc546c2761208743d1625b

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction repository in the bug description

**Comment by viceice** [comment]
Thats expected behavior when grouping different dep types. You need to manually define the commit type, as we don't garantee any order.
